src/ripple/json/impl/json_value.cpp

```diff
@@ -1068,6 +1068,24 @@ Value::get ( std::string const& key,
 }
 
 Value
+Value::removeElement ( std::uint32_t index )
+{
+    JSON_ASSERT (type_ == arrayValue || index < value_.map_->size ());
+
+    if ( type_ == nullValue || index >= value_.map_->size ())
```

Comment:
Neither of these conditions is possible - you eliminated them with an assert in the previous line!

My best guess:

    JSON_ASSERT ( type_ == nullValue  ||  type_ == arrayValue );
    if (type != arrayValue || index < 0 || index >= value_.map_.size())
        return null;